engines/gob/inter_v7.cpp

```diff
@@ -960,25 +1107,715 @@ void Inter_v7::o7_loadIFFPalette() {
+					break;
+				}
+
+				++htmlBufferPtr;
+			}
+		}
+
+		if (strcmp(htmlBufferPtr, commandArg.c_str()) == 0) {
+			_currentHtmlContext->_pos = _currentHtmlContext->_pos - strlen(_currentHtmlContext->_buffer) + htmlBufferPtr - _currentHtmlContext->_buffer;
+		} else if (!commandArg.empty()) {
+			warning("o7_seekHtmlFile(): Cannot find string \"%s\" while using SAVEMARK", commandArg.c_str());
+		}
+
+		if (_currentHtmlContext->_currentMarkIndex < 20) {
+			HtmlMark &mark = _currentHtmlContext->_marks[_currentHtmlContext->_currentMarkIndex];
+			mark._field_0 = READ_VARO_UINT32(destVar);
+			mark._field_4 = READ_VARO_UINT32(destVar + 4);
+			mark._field_8 = READ_VARO_UINT32(destVar + 8);
+			mark._field_C = READ_VARO_UINT32(destVar + 0xC);
+			mark._pos = _currentHtmlContext->_pos;
+			for (uint32 i = 0; i < 5; ++i) {
+				mark._field_14[i] = READ_VARO_UINT32(destVar + 0x14 + i * 4);
+			}
+
+			mark._field_28 = READ_VARO_UINT32(destVar + 0x28);
+
+			++_currentHtmlContext->_currentMarkIndex;
+		} else {
+			warning("o7_seekHtmlFile(): Mark index %d out of range while using SAVEMARK (max number of marks = %d)",
+					_currentHtmlContext->_currentMarkIndex,
+					20);
+		}
+
+	} else if (command.hasPrefix("GETMARK")) {
+		int markIndex = atoi(commandArg.c_str());
+		if (markIndex < 0 || markIndex >= (int) kMaxNbrOfHtmlMarks) {
+			warning("o7_seekHtmlFile(): mark index %d out of range", markIndex);
+			return;
+		}
+
+		_currentHtmlContext->_currentMarkIndex = markIndex;
+		HtmlMark &mark = _currentHtmlContext->_marks[markIndex];
+
+		_currentHtmlContext->_pos = mark._pos;
+
+		WRITE_VAR_OFFSET(destVar, mark._field_0);
+		WRITE_VAR_OFFSET(destVar + 4, mark._field_4);
+		WRITE_VAR_OFFSET(destVar + 8, mark._field_8);
+		WRITE_VAR_OFFSET(destVar + 0xC, mark._field_C);
+		for (uint32 i = 0; i < 5; ++i) {
+			WRITE_VAR_OFFSET(destVar + 0x14 + i * 4, mark._field_14[i]);
+		}
+
+		WRITE_VAR_OFFSET(destVar + 0x28, mark._field_28);
+
+	} else if (command == "SEEK") {
+		if (commandArg == "0")
+			_currentHtmlContext->_pos = 0;
+
+		if (_currentHtmlContext->_pos >= 0)
+			_currentHtmlContext->_stream->seek(_currentHtmlContext->_pos, SEEK_SET);
+		else
+			warning("o7_seekHtmlFile(): Invalid seek position %d", _currentHtmlContext->_pos);
+
+		// NB: Seek to other offsets than 0 does not seem to be implemented in the original engine
+
+	} else {
+		warning("o7_seekHtmlFile(): Unknown command \"%s\"", command.c_str());
+	}
+}
+
+Common::String Inter_v7::HtmlContext::popStringPrefix(const char **charPtr, char sep) {
+	const char *startPtr = *charPtr;
+	if (sep == ' ') {
+		while (!Common::isSpace(**charPtr) && **charPtr != '\0')
+			++*charPtr;
+		return Common::String(startPtr, *charPtr - startPtr);
+	} else {
+		while (**charPtr != sep && **charPtr != '\0')
+			++*charPtr;
+		return Common::String(startPtr, *charPtr - startPtr);
+	}
+}
+
+void Inter_v7::HtmlContext::parseTagAttributes(const char* tagAttributes) {
```

Comment:
```suggestion
void Inter_v7::HtmlContext::parseTagAttributes(const char *tagAttributes) {
```

I really wonder, if `common/formats/xmlparser.h` could be used instead.

Reply:
About reusing common/formats/xmlparser.h, I don't think it would really help. Gob's HTML parsing/rendering is quite funny, it is done partly by the games scripts, and partly by the engine through some low level  opcodes ("seek" and "parse attributes").